server/app/tool/readme.md

```diff
@@ -0,0 +1,119 @@
+
+## Supported TEEs
+
+Currently the `tool` supports the following types of TEE:
+- sgx
+- tdx
+- sample: The TEE used to demo/test Attestation Server's functionalities.
+
+## Commands
+
+The `tool` supports each TEE's `Policy(.rego)` and `Reference Data(.json)` configuration and testing subcommands, as the following:
+- get-policy
+- set-policy
+- get-reference-data
+- set-reference-data
+- restore-default-policy
+- restore-default-reference-data
+- test-attestation
+
+### get-policy
+
+Get the <TEE> specific evaluation `Policy(.regp)` from Attestation Server."
+```SHELL
+./tool get-policy --tee <TEE> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- addr(optional): Designate the attestation server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### set-policy
+
+Set the <TEE> specific evaluation `Policy(.regp)` according to the content of <POLICY>."
+```SHELL
+./tool set-policy --tee <TEE> --policy <POLICY> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- policy: The path of local `Policy(.rego)` which will be upload to Attestation Server.
+- addr(optional): Designate the Attestation Server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### get-reference-data
+
+Get the <TEE> specific `Reference Data(.json)` from Attestation Server.
+```SHELL
+./tool get-reference-data --tee <TEE> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- addr(optional): The Attestation Server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### set-reference-data
+
+Set the <TEE> specific evaluation `Reference Data(.json)` according to the content of <REFERENCE_DATA>.
+```SHELL
+./tool set-reference-data --tee <TEE> --reference-data <REFERENCE_DATA> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- reference-data: The path of local `Reference Data(.json)` which will be upload to Attestation Server.
+- addr(optional): Designate the Attestation Server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### restore-default-policy
+
+Restore the Attestation Server's <TEE> specific `Policy(.rego)` to default.
+```SHELL
+./tool restore-default-policy --tee <TEE> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- addr(optional): The Attestation Server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### restore-default-reference-data
+
+Restore the Attestation Server's <TEE> specific `Reference Data(.json)` to default.
+```SHELL
+./tool restore-default-reference-data --tee <TEE> [--addr <ADDR>]
+```
+- tee: Specify the target TEE name.
+- addr(optional): The Attestation Server's `management-sock` address. Default is `127.0.0.1:3001`.
+
+### test-attestation
+
+Test Attestation Server's `attestation` function with the input <EVIDENCE>.
+```SHELL
+./tool test-attestation [--evidence <EVIDENCE>] [--addr <ADDR>]
+```
+- evidence(optional): The evidence which is evaluated by Attestation Server. Default is a imbedded `sample` TEE's evidence.
```

Comment:
1. typo: a imbedded -> an embedded
2. what does the embedded sample evidence gives us? will anyone set sample policy or sample reference-data and would like to test them?

Reply:
This sample evidence is just a test case to test whether the base function of attestaion-server work correctly or not, the purpose of setting sample policy and reference data is only to confirm that these two "set" functions can work normally.

Reply:
Sample evidence is a built-in use case for testing the sample verifier of the attestation-service. We consider that users of this tool may only want to test that the basic function of the attestation-service work correctly (which is one of the purposes of this tool). Therefore, this sample evidence is built into the code of the tool to provide users with this function.

Reply:
If the built-in sample evidence is used with a built-in policy and built-in reference data, then we can simply do the same test during unit/integeration tests when building.

I think that the test-attestation command is useful for testing a user-defined policy; my question is -- is the built-in sample evidence useful for testing my own policy?

(please correct me if I'm missing a use case for test-attestation.)

Also -- maybe this command should be called simply `attest` ? (an imperative verb, like `get-policy`)